### DISCUSSION TITLE
Server middleware exclude

### DISCUSSION THREAD

**Discussion by mihail727** [nuxt/nuxt#17861] [Questions] [answered]
How to exclude middleware for a single request?
For example i have `auth` for validate user tokens:
`@/server/middleware/auth.ts`
Also i have login request:
`@/server/api/auth/login`
It is for this query that I would like to exclude

**Comment by mihail727** [comment]
I found this solution...
I form an array of exceptions in the middleware:
`@/server/middleware/auth.ts`
```ts
export default async (req: IncomingMessage, res: ServerResponse) => {

	const excludeURL = ['/auth/login'];

	if (!excludeURL.includes(req.url)) {
		// do something
	}
};
```
Is this considered a good solution or is there a better one?

**Comment by TheAlexLichter** [comment, marked as answer]
~~You can use the third function argument `next` to "skip" the current middleware. The type definition itself comes from [h3](https://github.com/unjs/h3/blob/main/src/handle.ts#L6) and is also used [in the framework](https://github.com/nuxt/framework/blob/main/packages/nitro/src/server/middleware.ts#L7)~~

Instead of the "old"  `next` syntax mentioned above, you can simply return inside the function 😋 

Untested example:

```ts
export default async (req: IncomingMessage, res: ServerResponse) => {
	const excludeURL = ['/auth/login'];

	if (!excludeURL.includes(req.url)) {
		return
	}
	// Some logic
};
```
